tools-local/unix/tool-helper.sh

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env bash
+
+# Provides helper functions.
+
+# Gets the repository root path.
+get_repo_root()
+{
+    scriptPath="$(cd "$(dirname "$0")"; pwd -P)"
+    repoRoot="$(cd "$scriptPath/../.."; pwd -P)"
+    echo "$repoRoot"
+}
+
+# Gets name of the operating system.
+# Exit 1 if unable to get operating system name.
+get_os_name()
+{
+    osName="$(uname -s)"
+
+    if [ $? -ne 0 ] || [ -z "$osName" ]; then
+        echo "Unable to determine the name of the operating system."
+        exit 1
+    fi
+
+    if echo "$osName" | grep -iqF "Darwin"; then
+        osName="OSX"
+    else
+        osName="Linux"
+    fi
+
+    echo "$osName"
+}
+
+# Gets the path to Tools/downloads folder under repository root.
+# Exit 1 if unable to determine the path.
+get_repository_tools_downloads_folder()
+{
+    repoRoot="$(get_repo_root)"
+    toolsPath="$repoRoot/Tools/downloads"
+    
+    if [ -z "$toolsPath" ]; then
+        echo "Unable to determine repository tools path."
+        exit 1
+    fi
+
+    echo "$toolsPath"
+}
+
+# Eval .toolversions file.
```

Comment:
I don't see the mechanism to override the `.toolversions` file with one outside the repo--is that a todo?